src/daft-dsl/src/expr/agg.rs

```diff
@@ -0,0 +1,63 @@
+use common_error::{DaftError, DaftResult};
+
+use crate::{AggExpr, ApproxPercentileParams, Expr, ExprRef};
+
+pub fn extract_agg_expr(expr: &ExprRef) -> DaftResult<AggExpr> {
```

Comment:
Larger change: Moved this out of daft-physical-plan (which we deleted) and put into daft-dsl since it is useful for both the local and ray runner.